src/main/java/com/onarandombox/MultiverseCore/listeners/MVPlayerListener.java

```diff
@@ -147,7 +149,13 @@ public void playerChangedWorld(PlayerChangedWorldEvent event) {
      */
     @EventHandler
     public void playerQuit(PlayerQuitEvent event) {
-        this.plugin.removePlayerSession(event.getPlayer());
+        Player player = event.getPlayer();
+        if (this.plugin.getMVPerms().hasPermission(player, "multiverse.teleport.self."
+                + LastLocationDestination.getID(), true)
```

Comment:
Theoretically the third arg here should be false, but it isn't used anymore so I guess it doesn't matter.